device/thunder_ripple_sdk/src/client/plugin_manager.rs

```diff
@@ -83,10 +83,11 @@ impl ThunderError {
 impl PluginStatus {
     pub fn to_plugin_state(&self) -> PluginState {
         match self.state.as_str() {
-            "activated" => PluginState::Activated,
-            "resumed" => PluginState::Activated,
-            "suspended" => PluginState::Activated,
+            "activated" | "resumed" | "suspended" => PluginState::Activated,
             "deactivated" => PluginState::Deactivated,
+            "deactivation" => PluginState::Deactivation,
+            "activation" | "precondition" => PluginState::Activation,
+            "unavailable" | "hibernated" => PluginState::Unavailable,
             _ => PluginState::Missing,
```

Comment:
MIssing is probably not a true measure for a Plugin State as Thunder responds with Error message for callsigns which are not there in the system

Reply:
Resolved it in the next commit